frame/ethereum/src/lib.rs

```diff
@@ -279,8 +279,17 @@ impl<T: Trait> Module<T> {
 		}
 
 		CurrentBlock::put(block.clone());
-		CurrentReceipts::put(receipts.clone());
-		CurrentTransactionStatuses::put(statuses.clone());
+		if receipts.len() > 0 {
+			CurrentReceipts::put(receipts.clone());
+		} else {
+			CurrentReceipts::kill();
+		}
+
+		if statuses.len() > 0 {
+			CurrentTransactionStatuses::put(statuses.clone());
+		} else {
+			CurrentTransactionStatuses::kill();
+		}
```

Comment:
How does this work? I thought these storage items should be cleared in `on_initialize`?

Reply:
> How does this work? I thought these storage items should be cleared in on_initialize?


So ugly isn't it? ^^ We need to move it somewhere else yes. Was just a way of get the thing working when I was doing the auxiliary logs stuff. 

Why is there: `receipts` and `statuses` vectors are initialized on top of the function and I want a `None` in those Storagevalues when there are no transactions in the current block. That way I can avoid inserting in them in auxiliary when I access this storagetdata from the frontier consensus.